path_login.go

```diff
@@ -333,7 +339,17 @@ func (b *kubeAuthBackend) parseAndValidateJWT(jwtStr string, role *roleStorageEn
 	}
 
 	// verify the namespace is allowed
-	if len(role.ServiceAccountNamespaces) > 1 || role.ServiceAccountNamespaces[0] != "*" {
+	valid := false
```

Comment:
I think you can make this `var valid bool`, since that will be false by default.